### PR TITLE
Update images and enable leak detection, and split tests in CI to Unit, Integration and Web

### PR DESCRIPTION
**Pull request by pawelchcki**
### Description

Use new Alpine based Debug images for most tests to ensure repeatable results. 

### Readiness checklist
- ~[ ] [Changelog entry](docs/changelog.md) added, if necessary~
- [x] Tests added for this feature/bug

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -256,200 +248,66 @@ jobs:
...
-      - <<: *IMAGE_DOCKER_DD_AGENT
-      - <<: *IMAGE_DOCKER_ELASTICSEARCH2
-      - <<: *IMAGE_DOCKER_HTTPBIN
-      - <<: *IMAGE_DOCKER_REDIS
-      - <<: *IMAGE_DOCKER_MEMCHACED
-      - <<: *IMAGE_DOCKER_MYSQL
-      - <<: *IMAGE_DOCKER_MONGODB
-      - <<: *IMAGE_DOCKER_REQUEST_REPLAYER
+    environment:
+      DDAGENT_HOSTNAME: 127.0.0.1
```
Comment (labbati) [inline .circleci/config.yml]: Shouldn't this be `DD_AGENT_HOST`

File: .circleci/config.yml [hunk 2]
```diff
@@ -547,45 +405,109 @@ workflows:
...
+      - integration_tests:
+          name: "Intg tests 56"
+          integration_testsuite: "test-integrations-56"
+          image_tag: "ddtrace_alpine_php-5.6-debug"
+      - integration_tests:
+          name: "Intg tests 70"
+          integration_testsuite: "test-integrations-70"
+          image_tag: "ddtrace_alpine_php-7.0-debug"
+      - integration_tests:
+          name: "Intg tests 71"
```
Comment (labbati) [inline .circleci/config.yml]: Please don't hate me 😞 what about `s/Intg/Integrations/g`

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-version: 2.0
+version: 2.1
```
Comment (labbati) [inline .circleci/config.yml]: I think we ended up eating the `composer test-integration` part (different from `composer test-integrations-XX`)

File: .circleci/config.yml (hunk 2 above)
Comment (pawelchcki) [reply .circleci/config.yml]: I agree with all my :heart: but  
I shortened it to make PHP version actually visible in Circle Ci UI. Which makes long names essentially useless

Comment (pawelchcki) [reply .circleci/config.yml]: image

Comment (pawelchcki) [reply .circleci/config.yml]: See `Static Analy...`
These changes actually started with `Integration tests 54` and even `Web integration tests 54` etc. 
But this turned into a bunch of 
`Integration tes...`
`Integration tes...`
`Integration tes...`
`Integration tes...`
`Integration tes...`
`Web integratio...`
`Web integratio...`
`Web integratio...`
`Web integratio...`
`Web integratio...`

I would have start with with version number but. The name of a job cannot start with a number in `2.1`.

Maybe something like `PHP 54 Integration tests` could work WDYT ?

Comment (pawelchcki) [reply .circleci/config.yml]: Hmm actually I like that more than current short forms. Let me quickly change.

Comment (pawelchcki) [reply .circleci/config.yml]: WDYT? 
image

File: .circleci/config.yml (hunk 3 above)
Comment (pawelchcki) [reply .circleci/config.yml]: Its there but its not immediately obvious. I remove the yaml reference in lieu of normal `run` devfinition to make it more obvious

File: .circleci/config.yml (hunk 2 above)
Comment (labbati) [reply .circleci/config.yml]: I sort of hate that I asked you to change that 😄! I am ok with this, but feel free to go back to the initial proposal or to abbreviate Integrations again if you prefer 👍

Comment (pawelchcki) [reply .circleci/config.yml]: You were right, I didn't like them either. But with naming I just usually try to use >first< thing that came to my mind,  turns out 2nd idea was much better :D

**Pull request merged**